compiler/noirc_evaluator/src/ssa/acir_gen/mod.rs

```diff
@@ -2268,11 +2312,9 @@ impl Context {
 
 // We can omit the element size array for arrays which don't contain arrays or slices.
 fn can_omit_element_sizes_array(array_typ: &Type) -> bool {
-    if array_typ.contains_slice_element() {
```

Comment:
This should have been `is_nested_slice()` but we already banned nested slices so I just removed it. When working with `AsSlice` we would omit the element sizes array. When it came time to access a slice created with `as_slice()` we would attempt to use the element sizes array which was never initialized correctly and we would go out of bounds.

Reply:
I pulled this out into here https://github.com/noir-lang/noir/pull/4528